lms/db/_locks.py

```diff
@@ -0,0 +1,35 @@
+from enum import Enum
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Session
+
+
+class LockType(int, Enum):
+    """
+    Identifies a type of resource for an advisory lock.
+
+    Advisory locks are identified by a `(type, object_id)` tuple, where both
+    are 32-bit integers. This enum provides values for `type`. The meaning of
+    `object_id` depends on the type.
+    """
+
+    OAUTH2_TOKEN_REFRESH = 1
+    """
+    Lock for an OAuth 2 token update.
+
+    The object ID is the `OAuth2Token.id` value for the token being updated.
+    """
+
+
+def try_advisory_txn_lock(db: Session, lock_type: LockType, id_: int) -> bool:
```

Comment:
I used `txn` as a shorthand for `transaction` here. Now that I look at this again, it should just use the word `transaction`.